src/NuGet.Clients/NuGet.CommandLine/Commands/Command.cs

```diff
@@ -128,13 +129,17 @@ public void Execute()
         /// </summary>
         protected void SetDefaultCredentialProvider()
         {
-            // Register an additional provider for the console specific application so that the user
-            // will be prompted if a proxy is set and credentials are required
-            var credentialProvider = new SettingsCredentialProvider(
-                new ConsoleCredentialProvider(Console),
-                SourceProvider,
-                Console);
-            HttpClient.DefaultCredentialProvider = credentialProvider;
+            var providers = new List<Credentials.ICredentialProvider>();
+            var pluginProviders = new PluginCredentialProviderBuilder(Settings).BuildAll();
+
+            providers.Add(new CredentialProviderAdapter(new SettingsCredentialProvider(NullCredentialProvider.Instance, SourceProvider, Console)));
```

Comment:
please break long lines, rule of thumb PRs on github should not force the reviewer to scroll